File: leet/0078-subnets/subnets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        if len(nums) == 0: return []
        
        ret = []
        stack = []
        
        def dfs(s: int):  
            ret.append([] + stack)
                
            if s == len(nums):
                return
            
            for i in range(s, len(nums)):
                stack.append(nums[i])
                dfs(i + 1)
                stack.pop()
                
        dfs(0)
        return ret

File: leet/0078-subnets/subnets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    let n = nums.length
    if(n == 0) return []

    let stack = []
    let ret = []

    let bt = function(i){
        ret.push(stack.concat())

        for (;i < n ; i++){
            stack.push(nums[i])
            bt(i + 1)
            stack.pop()
        }
    }

    bt(0)
    return ret
};
